src/spdx.ts

```diff
@@ -0,0 +1,24 @@
+import spdxParse from 'spdx-expression-parse'
+import {satisfies as spdxSatisfies} from '@onebeyond/spdx-license-satisfies'
+
+// TODO: Add support for NOASSERTION (#704), (#575)
+// TODO: Handle uknown licenses (#714)
+export function satisfies(license: string, expr: string): boolean {
+  return spdxSatisfies(license, expr)
+}
+
+export function isValid(license: string): boolean {
+  try {
+    spdxParse(license)
+    return true
+  } catch (e) {
+    // make a note in the log if the exception
+    // was caused by the expression parser, this
+    // usually means a discrepancy in ESM/CJS.
+    if (e instanceof TypeError) {
+      throw new Error('spdx: invalid expression parser')
+    }
```

Comment:
🤔 should the error message instead be `spdx: invalid expression` or something? the parser isn't the problem here, it's the inputs right?

Also under which error conditions is it safe to swallow the error and message here and just return `false` as in line 22?

Reply:
> should the error message instead be spdx: invalid expression or something? the parser isn't the problem here, it's the inputs right?

It's the parser in the sense that `TypeError` is what's raised when there's an error loading the parser. I have not seen these errors outside of local development, where they're common if you forget to `npm i` after bootstrapping the project.

> Also under which error conditions is it safe to swallow the error

None, if this fails we should abort the run.

Reply:
☝️ same here does that sync with your understanding @hmaurer ? we just bail if we can't parse a license, even if there may be vuln pkgs etc. found in this run?

again, I have little context/depth on DR Action at present, so if this is a non-concern, feel free to ignore 👍